includes/Core/Site_Health/Tag_Placement.php

```diff
@@ -171,14 +140,38 @@ protected function tags_placement_test() {
 			return $result;
 		}
 
+		if ( ! $this->environment_tag_guard->can_activate() ) {
+			$result['description'] = sprintf(
+				'<p>%s</p>',
+				__( 'Tags are not permitted in the current environment. They can only be placed in production mode.', 'google-site-kit' )
+			);
+
+			return $result;
+		}
+
 		$response = wp_remote_retrieve_body( $response );
 
 		$description = array();
 		foreach ( $active_modules as $module ) {
-			$tag_found = $this->check_if_tag_exists( $module, $response );
+			$settings = $module->get_settings()->get();
+			// If module has `canUseSnippet` setting, check if it is disabled.
+			if ( isset( $settings['canUseSnippet'] ) && empty( $settings['useSnippet'] ) ) {
```

Comment:
This is okay for now, but it could be simplified if we had access to the Tag instance, in which case we could check if it `can_register` which would check all of the tag's guards without needing to know what they are as we're doing here.

Reply:
Included in issue #8118